### PR TITLE
feat(pd): support gradient accumulation

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
support gradient accumulation for paddle backend.

## Summary by CodeRabbit

- **New Features**
  - Configurable gradient accumulation (acc_freq) that batches optimizer updates, optional gradient clipping, and multi‑GPU gradient sync to occur at the configured interval; acc_freq=1 preserves prior behavior.

- **Documentation**
  - Added argument docs and a Paddle backend notice describing acc_freq.

- **Tests**
  - Added tests exercising gradient accumulation and updated test cleanup.

### CODE REVIEW

**Comment by HydrogenSulfate** [comment]
> I am wondering that if you don't add this argument to `deepmd/utils/argcheck.py`, an error will be thrown during the argument check

Thanks for pointing this out. `size_t` is more appropriate than `int`: https://github.com/PaddlePaddle/Paddle/blob/53f2a48fd48d0f97eef53a0a88c1b79283b39b88/paddle/fluid/inference/api/paddle_api.h#L121%E2%80%9D

File: deepmd/utils/argcheck.py [hunk 1]
```diff
@@ -3167,6 +3168,7 @@ def training_args(
     doc_kf_blocksize = "The blocksize for the Kalman filter."
     doc_model_prob = "The visiting probability of each model for each training step in the multi-task mode."
     doc_data_dict = "The multiple definition of the data, used in the multi-task mode."
+    doc_acc_freq = "The accumulation steps for the gradients."
```
Comment (njzjz) [inline deepmd/utils/argcheck.py]: Isn't it frequency?

**Review by njzjz — APPROVED** [review]
